Mathlib/Data/Matroid/Circuit.lean

```diff
@@ -0,0 +1,173 @@
+/-
+Copyright (c) 2025 Peter Nelson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Peter Nelson
+-/
+import Mathlib.Data.Matroid.Closure
+
+/-!
+# Matroid Circuits
+
+A `Circuit` of a matroid `M` is a minimal set `C` that is dependent in `M`.
+A matroid is determined by its set of circuits, and often the circuits
+offer a more compact description of a matroid than the collection of independent sets or bases.
+In matroids arising from graphs, circuits correspond to graphical cycles.
+
+# Main Declarations
+
+* `Matroid.Circuit M C` means that `C` is minimally dependent in `M`.
+-/
+
+variable {α : Type*} {M : Matroid α} {C C' I X Y R : Set α} {e f x y : α}
+
+open Set
+
+namespace Matroid
+
+/-- A `Circuit` of `M` is a minimal dependent set in `M` -/
+def Circuit (M : Matroid α) := Minimal M.Dep
```

Comment:
```suggestion
def IsCircuit (M : Matroid α) : Prop := Minimal M.Dep
```
Since this is a propositional statement, a name like `IsCircuit` would be more typical

Reply:
Is the point that `Circuit` is a noun? The matroid predicates defined so far (`Base`, `Basis`, `Indep`) don't have `Is`, and this aligns with (for example) `LinearIndependent` and `Set.Finite`. 

If so, I don't think there's really a danger of ambiguity, and I'm worried about the cost in keystrokes.

Reply:
Is it recommended that I apply that here? With matroid props, I don't love having it for some and not others.

Reply:
Ok, then let's merge as is. Then you can switch over the entire API in a future PR.

Reply:
I think in some cases the current matroid convention makes it hard to guess what a name is about: for example it would be a lot clearer if [Matroid.Restriction](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Matroid/Restrict.html#Matroid.Restriction) is renamed to `Matroid.IsRestrict`.